### PR TITLE
fix: make git_cherry_pick() accept a shortened hash

### PR DESCRIPTION
**Pull request by maelle**
I tried using `resolve_refish(ref, repo)` in rebase.c but I got stuck because I wasn't able to create a thing of the right type for this line:

https://github.com/r-lib/gert/blob/d0fc50f033dda9e8f9634d20fe47462cb188f40b/src/rebase.c#L92

I'd be curious to see how to actually solve this on the C side.

### CODE REVIEW

File: R/rebase.R [hunk 1]
```diff
@@ -83,6 +83,9 @@ git_reset <- function(type = c("soft", "hard", "mixed"), ref = "HEAD", repo = ".
 #' @useDynLib gert R_git_cherry_pick
 #' @param commit id of the commit to cherry pick
 git_cherry_pick <- function(commit, repo = '.'){
+  if (!is_full_hash(commit)) {
```
Comment (maelle) [inline R/rebase.R]: I got the idea from

https://github.com/r-lib/gert/blob/d0fc50f033dda9e8f9634d20fe47462cb188f40b/R/submodules.R#L48

**Comment by jeroen** [comment]
To fix the check you need to move `git_config_set()` down and give it the `repo` paramater. We are not allowed to modify global git configuration settings on CRAN so therefore they need to be set locally per repo.

Alternatively you can use `git_commit(...., author = "jerry <jerry@gmail.com>")`

**Comment by jeroen** [comment]
Thanks !

**Comment by maelle** [comment]
thank you!!

**Pull request merged**